File: app/src/main/java/com/pluu/webtoon/ui/view/AspectRatioImageView.java
package com.pluu.webtoon.ui.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

/**
 * Width Fix, Height Ratio ImageView
 * Created by PLUUSYSTEM-NEW on 2016-01-28.
 */
public class AspectRatioImageView extends ImageView {

    private float hRatio = -1;

    public AspectRatioImageView(Context context) {
        super(context);
    }

    public AspectRatioImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AspectRatioImageView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        if (hRatio == -1) {
            setMeasuredDimension(widthMeasureSpec, heightMeasureSpec);
        } else {
            int width = MeasureSpec.getSize(widthMeasureSpec);
            int height = (int) (width * hRatio);
            setMeasuredDimension(width, height);
        }
    }

    public void sethRatio(float hRatio) {
        this.hRatio = hRatio;
    }
}

File: app/src/main/java/com/pluu/webtoon/ui/view/AspectRatioImageView.kt
package com.pluu.webtoon.ui.view

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.util.AttributeSet
import android.view.View

/**
 * Width Fix, Height Ratio ImageView
 * Created by pluu on 2017-05-26.
 */
class AspectRatioImageView constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : AppCompatImageView(context, attrs, defStyle) {

    private var hRatio = -1f

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        if (hRatio == -1f) {
            setMeasuredDimension(widthMeasureSpec, heightMeasureSpec)
        } else {
            val width = View.MeasureSpec.getSize(widthMeasureSpec)
            val height = (width * hRatio).toInt()
            setMeasuredDimension(width, height)
        }
    }

    fun sethRatio(hRatio: Float) {
        this.hRatio = hRatio
    }
}
